### ISSUE TITLE
Filtering the test case by browser name and OS

### ISSUE THREAD

**Issue by perwezalamnewscorp** [renatoathaydes/spock-reports#129]
When I am running my test in multiple browser sequentially using Gradle tool. In this case test will be runing in all the configured browsers and then generate the final report. If you see the default report of GEB, It has segregated report folder for all the browser.

Now If you see in Spock-report, It has consolidated report. Its good. But when i navigate to any report, then I should see , which test has run in which browser and should be able to filter it as well.

I dont see, there is way for creating separate report for each browser when test is running sequentially or in parallel. It means there should have some way for adding these info into the report and should be able to filter the test cases by this. I thought it could be easily achieved by adding the browser info into the report and the same can be used to filter the test

As you had said, This should be handled by test developer to generate the report separately for all browsers.

How could i achieve this ? Please help me to understand this.

**Comment by renatoathaydes** [comment]
I would suggest just changing the directory where the reports are created when you have the same tests running over and over again using different environments (in your case, browser, but maybe also things like different JVM versions).

Would that work for you?

**Comment by perwezalamnewscorp** [comment]
Since my test runs in parallel or sequentially in the multiple browsers(Chrome.Firefox and safari). In this cases I am not sure how could we create the separate directory for each platform (Browser or JVM or OS) and generate the report. Currently we do not need to write any code to generate report of SPOCK, It is taken care by plugins itself. I think this solution is bit complicated and also user always expect the consolidated report of all test Platform instead of separate report for each.

I have used many test report creator plugins such as Jasmin reporter(Nodejs) and Extent report using TestNG or JUnit. Here user can provide the category(browser name or OS or JVM or otherinfo) at runtime for each tests and in the final report, It shows the filter for this. User can filter test report by category whatever it is. Its very easy for end user to implement this in test framework.

Parallel testing in Web app is general use cases. I strongly believe to have some way to add filter option into the report using easy implementation rather than having created separate directory.

I think we can do  something like this in spock-report in generic way.

     def setupSpec(){
        addCategory(driver.browser.name)
        addCategory('OsName')
      }
this will add the category for each spec file and same can be displayed into the report. I think this solution should work for Java App test as well. This will be very useful feature


Here is the link of Extent report (See the screen shot with category)

http://extentreports.com/docs/versions/3/java/
<img width="925" alt="screen shot 2017-12-13 at 2 27 28 pm" src="https://user-images.githubusercontent.com/19994714/33930099-cfca2e34-e011-11e7-8dba-65aea46d6b6b.png">

**Comment by renatoathaydes** [comment]
A guiding principle of Spock-Reports is that HTML reports are completely static, so you can print it and open it in any browser without JS.

So this is out of scope for the default HTML report. However, feel free to contribute an alternative Report creator that can do what you want.
